src/main/java/uk/gov/companieshouse/api/accounts/service/impl/AbstractServiceImpl.java

```diff
@@ -37,23 +43,32 @@ public AbstractServiceImpl(MongoRepository<U, String> mongoRepository,
     }
 
     @Override
-    public ResponseObject<T> create(T rest, String companyAccountId) {
+    public ResponseObject<T> create(T rest, Transaction transaction, String requestId)
+            throws DataException {
+        String companyNumber = transaction.getCompanyNumber();
+
         addEtag(rest);
         addKind(rest);
         U baseEntity = genericTransformer.transform(rest);
 
-        try{
-        baseEntity.setId(generateID(companyAccountId));
+        final Map<String, Object> debugMap = new HashMap<>();
+        debugMap.put("transaction_id", transaction.getId());
+        debugMap.put("company_accounts_id", companyNumber);
```

Comment:
this isn't the company_accounts_id, we shouldn't be using companyNumber here at all